implementation.js

```diff
@@ -1,6 +1,102 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var has = require('has');
+var global = require('system.global')();
+
+var parseIterable = function (iterator) {
+	var done = false;
+	var iterableResponse;
+	var tempArray = [];
+
+	if (iterator && typeof iterator.next === 'function') {
+		while (!done) {
+			iterableResponse = iterator.next();
+			if (
+				has(iterableResponse, 'value')
+				&& has(iterableResponse, 'done')
+			) {
+				if (iterableResponse.done === true) {
+					done = true;
+					break; // eslint-disable-line no-restricted-syntax
+
+				} else if (iterableResponse.done !== false) {
+					break; // eslint-disable-line no-restricted-syntax
+				}
+
+				tempArray.push(iterableResponse.value);
+			} else if (iterableResponse.done === true) {
+				done = true;
+				break; // eslint-disable-line no-restricted-syntax
+			} else {
+				break; // eslint-disable-line no-restricted-syntax
+			}
+		}
+	}
+
+	return done ? tempArray : false;
+};
+
+var hasSymbols = typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol';
+var iteratorSymbol;
+var forOfOnly;
+if (hasSymbols) {
+	iteratorSymbol = Symbol.iterator;
+} else {
+	var supportsStrIterator = (function () {
+		try {
+			var supported = false;
+			var obj = { // eslint-disable-line no-unused-vars
+				'@@iterator': function () {
+					return {
+						'next': function () {
+							supported = true;
+							return {
+								'done': true,
+								'value': undefined
+							};
+						}
+					};
+				}
+			};
+
+			Function('obj', // eslint-disable-line no-new-func
+				'for (var x of obj) {}'
+			)(obj);
+			return supported;
+		} catch (e) {
+			return false;
+		}
+	}());
+
+	if (supportsStrIterator) {
+		iteratorSymbol = '@@iterator';
+	} else if ('Set' in global && 'Map' in global) {
+		var map = new Map();
+		map.set(1, 2);
+		var y;
+		Function('map', 'y', 'for (var x of map) {y = x}')(map, y); // eslint-disable-line no-new-func
+		if (y[0] === 1 && y[1] === 2) {
+			forOfOnly = true;
+		}
+	}
+}
+
+var isCallable = require('is-callable');
+
+var usingIterator = function (items) {
+	if (has(items, iteratorSymbol)) {
+		return items[iteratorSymbol]();
+	} else if ('Set' in global && 'Map' in global && isCallable(items.entries) && isCallable(items.values)) {
```

Comment:
Indeed they are needed, because in some engines, there are no Symbols, no methods that produce iterators, but there is Set/Map, and they're iterable with `for..of`.